src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -340,3 +340,20 @@ void Ekf::predictState(const imuSample &imu_delayed)
 	float alpha_height_rate_lpf = 0.1f * imu_delayed.delta_vel_dt; // 10 seconds time constant
 	_height_rate_lpf = _height_rate_lpf * (1.0f - alpha_height_rate_lpf) + _state.vel(2) * alpha_height_rate_lpf;
 }
+
+void
+Ekf::resetGlobalPosToExternalObservation(double lat_deg, double lon_deg, float accuracy, uint64_t timestamp_observation) {
+
+   if (!_pos_ref.isInitialized()) {
+       return;
+   }
+
+   // apply a first order correction using velocity at the delated time horizon and the delta time
+   timestamp_observation = min(_time_latest_us, timestamp_observation);
```

Comment:
```suggestion
   timestamp_observation = math::min(_time_latest_us, timestamp_observation);
```

Should we maybe reject the observation if it is more recent than current time or outdated?

Reply:
@bresch Actually, I just noticed that we are constraining the timestamp between the timestamp of the vehicle command (which is a PX4 internal command) and that timestamp minus kMaxDelaySecondsExternalPosMeasurement.
So I think it should actually not be possible to be more recent (or at least not by much) than the latest IMU data.
Unless you want to push those constraints into the EKF, which I would only do it it's explicit, e.g. by parameter.
What do you think?

Reply:
@bresch Any opinion, I addressed the other comments.

Reply:
I think that's good enough